aiida/tools/importexport/dbexport/__init__.py

```diff
@@ -29,31 +29,34 @@
 )
 from aiida.tools.importexport.common.utils import export_shard_uuid
 from aiida.tools.importexport.dbexport.utils import (
-    check_licenses, fill_in_query, serialize_dict, check_process_nodes_sealed
+    check_licenses, fill_in_query, serialize_dict, check_process_nodes_sealed, summary, EXPORT_LOGGER
 )
 
 from .zip import ZipFolder
 
-__all__ = ('export', 'export_zip')
+__all__ = ('export',)
 
 
-def export_zip(what, outfile='testzip', overwrite=False, silent=False, use_compression=True, **kwargs):
-    """Export in a zipped folder
+def export(entities, filename=None, file_format=None, overwrite=False, silent=False, use_compression=True, **kwargs):
```

Comment:
Note that you are breaking backwards compatibility here by removing the `export_zip` function and by changing the argument `outfile` to `filename`. I checked the [list of public resources](https://aiida.readthedocs.io/projects/aiida-core/en/latest/working_with_aiida/python_api.html) on the documentation, and neither `export` nor `export_zip` are on there to my surprise, so technically we are in the clear here. Just something we may want to think about a bit

Reply:
I know. And I can change it back if needed. But when/if we update this module, this will all change anyway.
And while it looks like I'm removing `export_zip` here, it's not true. I am merely renaming the former function `export` to `export_tar` and creating a wrapper function `export` to call either `export_zip` or `export_tar`, giving it a proper public API - instead of the weird setup that was the case before, where there were three major function one could call:

- `export_tree`: In reality not a small function that tries to export into a system folder, but the actual main export function.
- `export_zip`: A wrapper for `export_tree` to stream it all into a home-backed `ZipFolder` class (and subsequently a zip file).
- `export`: A wrapper for `export_tree` to run it and later tar and gzip it, resulting in a `tar.gz` file.

Now we have instead the following:

- `export_tree`: No changes, still the major export function.
- `export_zip`: No changes, other than it now focuses on what it should do, i.e., run main export function and zip the result.
- `export_tar`: Renamed `export`, which, like `export_zip`, now focuses on what it sohuld do, i.e., run main export function and tar and gzip the result.
- ´export`: Wrapper function with logging and timings for `export_tar` and `export_zip`.

Reply:
Thanks for the clarification. Since people may indeed have used the `export` function in scripts (I certainly know that we have used `export_zip` in scripts), could you make it such that the change is still backwards compatible?
I.e. that a call to `export` (e.g. using `outfile`) with the old parameters will continue to work (I don't really care whether it produces a tar or a zip; feel free to decide the default).